src/macos.rs

```diff
@@ -111,14 +95,135 @@ impl Sandbox for MacSandbox {
     }
 }
 
+impl MacSandbox {
+    /// Add or modify a path's exceptions.
+    fn update_path_exceptions(&mut self, path: PathBuf, exceptions: PathException) -> Result<()> {
+        // Canonicalize all exception paths.
+        //
+        // Since the macOS sandbox only cares about permissions for symlink targets, due
+        // to the `(allow file-read-metadata)` rule, we don't need to bother with
+        // keeping the original paths.
+        let escaped_path = escape_path(&path)?;
+
+        let exception = self.path_exceptions.entry(escaped_path).or_insert(PathException::empty());
+        exception.insert(exceptions);
+
+        Ok(())
+    }
+
+    /// Create a seatbelt profile for the requested sandbox configuration.
+    fn create_profile(&self) -> Result<Vec<u8>> {
+        let mut profile = DEFAULT_RULE.to_vec();
+
+        // Sort by component count to ensure parent paths appear before descendants.
+        let mut path_exceptions: Vec<_> = self.path_exceptions.iter().collect();
+        path_exceptions.sort_unstable_by(|a, b| a.0.len().cmp(&b.0.len()));
+
+        for (path, exception) in path_exceptions {
+            // Deny all access to clear existing permission grants.
+            Self::revoke_path_access(&mut profile, path)?;
```

Comment:
I think you only need to revoke the accesses that we aren't allowing. So you could generate a more efficient profile by doing a bitwise not of the exceptions and specifically revoking those.

```rs
            // Deny access to anything not granted
            Self::revoke_path_access(&mut profile, path, exceptions.complement())?;
```

But it also works if you revoke everything, so I guess that shouldn't matter too much.

Reply:
Yeah my goal was primarily to avoid complexity because it simplifies testing, review, and maintenance.

Realistically I don't think the size of the profiles matters much.